src/main/java/org/folio/circulation/domain/UpdateRequestQueue.java

```diff
@@ -129,13 +149,50 @@ private CompletableFuture<Result<Request>> awaitPickup(Request request) {
             populateHoldShelfExpirationDate(
               request.withPickupServicePoint(servicePoint),
               tenantTimeZone
-            ))
+            ).map(calculatedRequest-> setHoldShelfExpirationDateWithExpirationDateManagement(tenantTimeZone, calculatedRequest,
+              requestQueue, originalRequest)))
         );
     } else {
-      return completedFuture(succeeded(request));
+      Request updatedRequest = Request.from(request.asJson());
+      requestQueue.update(originalRequest, updatedRequest);
+
+      return requestRepository.update(request)
+        .thenComposeAsync(result -> result.after(v -> requestQueueRepository.updateRequestsWithChangedPositions(requestQueue)));
     }
   }
 
+  private RequestQueue setHoldShelfExpirationDateWithExpirationDateManagement(ZoneId tenantTimeZone, Request calculatedRequest,
+                                                                                RequestQueue requestQueue, Request originalRequest) {
+
+    ExpirationDateManagement expirationDateManagement = calculatedRequest.getPickupServicePoint().getHoldShelfClosedLibraryDateManagement();
+    String intervalId = calculatedRequest.getPickupServicePoint().getHoldShelfExpiryPeriod().getIntervalId().toUpperCase();
+    // Old data where strategy is not set so default value but TimePeriod has MINUTES / HOURS
+    if (ExpirationDateManagement.KEEP_THE_CURRENT_DUE_DATE.name().equals(expirationDateManagement.name()) && isShortTerm(intervalId)) {
+      expirationDateManagement = ExpirationDateManagement.KEEP_THE_CURRENT_DUE_DATE_TIME;
+    }
+
+    ExpirationDateManagement finalExpirationDateManagement = expirationDateManagement;
+
+    calendarRepository.lookupOpeningDays(calculatedRequest.getHoldShelfExpirationDate().toLocalDate(),
+        calculatedRequest.getPickupServicePoint().getId())
+      .thenApply(adjacentOpeningDaysResult -> determineClosedLibraryStrategyForHoldShelfExpirationDate(
+        finalExpirationDateManagement, calculatedRequest.getHoldShelfExpirationDate(),
+        tenantTimeZone, calculatedRequest.getPickupServicePoint().getHoldShelfExpiryPeriod()
+      ).calculateDueDate(calculatedRequest.getHoldShelfExpirationDate(), adjacentOpeningDaysResult.value()))
+      .thenApply(calculatedDate -> {
+        calculatedRequest.changeHoldShelfExpirationDate(calculatedDate.value());
+        requestQueue.update(originalRequest,calculatedRequest);
```

Comment:
This block would be more readable if you defined the `ClosedLibraryStrategy` outside of the chained like this:
```
ExpirationDateManagement finalExpirationDateManagement = expirationDateManagement;
ClosedLibraryStrategy closedLibraryStrategy = determineClosedLibraryStrategyForHoldShelfExpirationDate
    (finalExpirationDateManagement, calculatedRequest.getHoldShelfExpirationDate(), tenantTimeZone, calculatedRequest.getPickupServicePoint().getHoldShelfExpiryPeriod());

calendarRepository.lookupOpeningDays(calculatedRequest.getHoldShelfExpirationDate().toLocalDate(), calculatedRequest.getPickupServicePoint().getId())
  .thenApply(adjacentOpeningDaysResult -> closedLibraryStrategy.calculateDueDate(calculatedRequest.getHoldShelfExpirationDate(), adjacentOpeningDaysResult.value()))
  .thenApply(calculatedDate -> {
    calculatedRequest.changeHoldShelfExpirationDate(calculatedDate.value());
    requestQueue.update(originalRequest,calculatedRequest);

    return requestRepository.update(calculatedRequest)
      .thenComposeAsync(result -> result.after(v -> requestQueueRepository.updateRequestsWithChangedPositions(requestQueue)));
  });
```

Reply:
added